src/main/scala/is/hail/HailContext.scala

```diff
@@ -449,12 +439,14 @@ class HailContext private(val sc: SparkContext,
     checkDatasetSchemasCompatible(vsms, inputs)
 
     // I can't figure out how to write this with existentials -cs
-    if (vsms(0).isGenericGenotype) {
-      val gdses = vsms.asInstanceOf[Array[GenericDataset]]
+    if (vsms(0).genotypeSignature == TGenotype) {
```

Comment:
Yes, good catch.  This builds on https://github.com/hail-is/hail/pull/2039, I will fix it there.